php-frontend/src/main/java/org/sonar/php/tree/symbols/SymbolTableImpl.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010 SonarSource and Akram Ben Aissi
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.php.tree.symbols;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import org.sonar.plugins.php.api.symbols.Symbol;
+import org.sonar.plugins.php.api.symbols.SymbolTable;
+import org.sonar.plugins.php.api.tree.CompilationUnitTree;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.expression.IdentifierTree;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+public class SymbolTableImpl implements SymbolTable {
+
+  private List<Symbol> symbols = new ArrayList<>();
+  private Set<Scope> scopes = Sets.newHashSet();
+
+  private SymbolTableImpl(){
+  }
+
+  public static SymbolTableImpl create(CompilationUnitTree compilationUnit) {
+    SymbolTableImpl symbolModel = new SymbolTableImpl();
+    new SymbolVisitor(symbolModel).visitCompilationUnit(compilationUnit);
+    return symbolModel;
+  }
+
+  public void addScope(Scope scope){
+    scopes.add(scope);
+  }
+
+  @Override
+  public ImmutableSet<Scope> getScopes(){
+    return ImmutableSet.copyOf(scopes);
```

Comment:
@vilchik-elena Do we really need to copy this set? Couldn't wrap it in a read-only Set e.g. with java.util.Collections.unmodifiableSet(set)?